### DISCUSSION TITLE
NSpainel Detach switches from relays and send MQTT

### DISCUSSION THREAD

**Discussion by tv4you2016** [arendst/Tasmota#19037] [Q&A]
hello,
I have an Nspanel updated with tasmota tasmota32-nspanel.bin (Tasmota v13.0.0 Qasim)
with the  template:

**{"NAME":"NSPanel","GPIO":[0,0,0,0,3872,0,0,0,0,0,32,0,0,0,0,225,0,480,224,1,0, 0,0,33,0,0,0,0,0,0,0,0,0,0,4736,0],"FLAG":0,"BASE":1}**

- used the command **SetOption114 1** for all switches to be disconnected from their respective relays.

- I connected by mqtt to HomeAssistant, and the device is detected correctly, showing 2 outs and the temperature sensor,

the 2 buttons just don't appear even when clicking on the automation triggers

can someone help me how to put the buttons to appear in the HA?

![image](https://github.com/arendst/Tasmota/assets/83761813/4cb53ee0-1c67-429b-a4a5-d0bad05a2c2f)

![image](https://github.com/arendst/Tasmota/assets/83761813/19d976e3-0529-400c-813f-23dba4ca069f)

**Comment by sfromis** [comment]
As the template has no Switch pins, using SetOption114 has no effect. The similar one for Button pins (which the template has two of) is SetOption73.

**Reply by tv4you2016** [reply]
OK ..
I ended up using the SetOption73 1 command, and it still doesn't appear in HA

**Reply by sfromis** [reply]
The console will show you what Tasmota is doing, including what MQTT payloads are published. How to "convince" [HA](https://tasmota.github.io/docs/Home-Assistant/) to do what you'd like with those messages is outside of what I've chosen to know about.

**Reply by tv4you2016** [reply]
![image](https://github.com/arendst/Tasmota/assets/83761813/ed2e3347-0b0d-4441-8c73-e9ba382ae733)

**Reply by sfromis** [reply]
Waste of time to post a screenshot, and expect people to understand what you were doing.

**Reply by tv4you2016** [reply]
I used the SetOption 73 1 command and it still doesn't appear

**Reply by sfromis** [reply]
You are still telling no useful info about what you were doing (or not doing). Activating SetOption73 merely changes what will happen going forward, expect no immediate output after having changed the setting, only when you start to do something with the buttons.

**Reply by tv4you2016** [reply]
what i want is:
 disconnect the 2 physical buttons from their respective relays.
and those same buttons appear in the HA so I can do whatever I want with them

**Reply by tv4you2016** [reply]
for this it is necessary that the buttons appear on the HA something that is not happening

**Reply by sfromis** [reply]
I can't tell you about how to wrestle with HA, but a key is that you first make sure that the info is published, by checking the logs. And "checking" does not mean posting screenshots without context.

**Reply by tv4you2016** [reply]
Sorry if I wasn't clear.

I see that in the console the events are sent. I marked in red.
but in HA it does not appear

![251167534-4cb53ee0-1c67-429b-a4a5-d0bad05a2c2f](https://github.com/arendst/Tasmota/assets/83761813/41f07750-34b4-4b4b-8066-55e4aa3f398f)

**Reply by tv4you2016** [reply]
and what I want to do is:
that the 2 physical buttons of the NSPanel appear in the HA

**Reply by sfromis** [reply]
Ok, now you've verified that SetOption73 does exactly what it is supposed to do, publishing payloads available to HA to do something with. Did you restart after activating the option? At least Tasmota, but _maybe_ also HA.

**Reply by tv4you2016** [reply]
Yes, I already rebooted HA and NSpanel and it didn't work, it still doesn't stop on HA

**Reply by tv4you2016** [reply]
I don't know if what I'm going to say makes sense but, using the same template mentioned above, I only removed the relays, the buttons appear in the HA, as shown in the image posted below

![image](https://github.com/arendst/Tasmota/assets/83761813/bf14dd55-298c-4075-9b1b-ce7bb286ffff)

in the console in the /config topic the following data is sent

` tasmota/discovery/E0E2E60BDB74/config = {"ip":"192.168.1.116","dn":"Tasmota","fn":["Tasmota",null,null,null,null,null,null,null] ,"hn":"tasmota-0BDB74-7028","mac":"E0E2E60BDB74","md":"NSPanel","ty":0,"if":0,"ofln":"Offline"," onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.0.0.1","t":"tasmota_0BDB74","ft" :"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[0,0,0,0,0,0,0,0 ,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], "swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1, -1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null ,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null ],"btn":[1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0 ,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0 ,"20":0,"30":0,"68":0,"73":1,"82":0,"114":0,"117":0},"lk":0, "lt_st":0,"sho":[],"sht":[],"ver":1} (retained)`


then I go back to the template and put the relays again and I see that in the /config topic the following data is sent:

`tasmota/discovery/E0E2E60BDB74/config = {"ip":"192.168.1.116","dn":"Tasmota","fn":["Tasmota","Tasmota2",null,null,null,null,null,null],"hn":"tasmota-0BDB74-7028","mac":"E0E2E60BDB74","md":"NSPanel","ty":0,"if":0,"ofln":"Offline","onln":"Online","state":["OFF","ON","TOGGLE","HOLD"],"sw":"13.0.0.1","t":"tasmota_0BDB74","ft":"%prefix%/%topic%/","tp":["cmnd","stat","tele"],"rl":[1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"swc":[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],"swn":[null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null,null],"btn":[1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"so":{"4":0,"11":0,"13":0,"17":0,"20":0,"30":0,"68":0,"73":1,"82":0,"114":0,"117":0},"lk":0,"lt_st":0,"sho":[],"sht":[],"ver":1} (retained)
`
now the Buttons do not appear, as shown in the following image

![image](https://github.com/arendst/Tasmota/assets/83761813/541b20c6-1b82-49b5-97c1-eb7678b0c79c)



@sfromis

**Reply by sfromis** [reply]
As already mentioned, asking _me_ about HA is well outside of where I have anything to add.

**Reply by barbudor** [reply]
AFAIK detached switches are not reported to HA in the discovery message.
You have to configure HA manually.
Something about a 'binary sensor' if I recall what I've seen in other posts
But I'm not a HA user neither
